src/devices/bus/multibus/isbc202.cpp

```diff
@@ -497,34 +497,34 @@ void isbc202_device::device_add_mconfig(machine_config &config)
 	}
 
 	// Connect CO/CI signals
-	m_mcu->fo_w().set(m_cpes[ 0 ] , FUNC(i3002_device::ci_w));
-	m_cpes[ 0 ]->co_w().set(m_cpes[ 1 ] , FUNC(i3002_device::ci_w));
-	m_cpes[ 1 ]->co_w().set(m_cpes[ 2 ] , FUNC(i3002_device::ci_w));
-	m_cpes[ 2 ]->co_w().set(m_cpes[ 3 ] , FUNC(i3002_device::ci_w));
-	m_cpes[ 3 ]->co_w().set(FUNC(isbc202_device::co_w));
+	m_mcu->set_fo_w_cb(m_cpes[ 0 ] , FUNC(i3002_device::ci_w));
+	m_cpes[ 0 ]->set_co_w_cb(m_cpes[ 1 ] , FUNC(i3002_device::ci_w));
+	m_cpes[ 1 ]->set_co_w_cb(m_cpes[ 2 ] , FUNC(i3002_device::ci_w));
+	m_cpes[ 2 ]->set_co_w_cb(m_cpes[ 3 ] , FUNC(i3002_device::ci_w));
+	m_cpes[ 3 ]->set_co_w_cb(FUNC(isbc202_device::co_w));
 
 	// Connect RO/LI signals
-	m_cpes[ 0 ]->ro_w().set(FUNC(isbc202_device::co_w));
-	m_cpes[ 1 ]->ro_w().set(m_cpes[ 0 ] , FUNC(i3002_device::li_w));
-	m_cpes[ 2 ]->ro_w().set(m_cpes[ 1 ] , FUNC(i3002_device::li_w));
-	m_cpes[ 3 ]->ro_w().set(m_cpes[ 2 ] , FUNC(i3002_device::li_w));
+	m_cpes[ 0 ]->set_ro_w_cb(FUNC(isbc202_device::co_w));
+	m_cpes[ 1 ]->set_ro_w_cb(m_cpes[ 0 ] , FUNC(i3002_device::li_w));
+	m_cpes[ 2 ]->set_ro_w_cb(m_cpes[ 1 ] , FUNC(i3002_device::li_w));
+	m_cpes[ 3 ]->set_ro_w_cb(m_cpes[ 2 ] , FUNC(i3002_device::li_w));
 
 	// Connect M-bus
-	m_cpes[ 0 ]->mbus_r().set([this]() { return mbus_r(); });
-	m_cpes[ 1 ]->mbus_r().set([this]() { return mbus_r() >> 2; });
-	m_cpes[ 2 ]->mbus_r().set([this]() { return mbus_r() >> 4; });
-	m_cpes[ 3 ]->mbus_r().set([this]() { return mbus_r() >> 6; });
+	m_cpes[ 0 ]->set_mbus_r_cb([this]() { return mbus_r(); } , "");
+	m_cpes[ 1 ]->set_mbus_r_cb([this]() { return mbus_r() >> 2; } , "");
+	m_cpes[ 2 ]->set_mbus_r_cb([this]() { return mbus_r() >> 4; } , "");
+	m_cpes[ 3 ]->set_mbus_r_cb([this]() { return mbus_r() >> 6; } , "");
 
 	// Connect I-bus
-	m_cpes[ 0 ]->ibus_r().set([this]() { return ibus_r(); });
-	m_cpes[ 1 ]->ibus_r().set([this]() { return ibus_r() >> 2; });
-	m_cpes[ 2 ]->ibus_r().set([this]() { return ibus_r() >> 4; });
-	m_cpes[ 3 ]->ibus_r().set([this]() { return ibus_r() >> 6; });
+	m_cpes[ 0 ]->set_ibus_r_cb([this]() { return ibus_r(); } , "");
+	m_cpes[ 1 ]->set_ibus_r_cb([this]() { return ibus_r() >> 2; } , "");
+	m_cpes[ 2 ]->set_ibus_r_cb([this]() { return ibus_r() >> 4; } , "");
+	m_cpes[ 3 ]->set_ibus_r_cb([this]() { return ibus_r() >> 6; } , "");
 
 	// Connect SX input
-	m_mcu->sx_r().set([this]() { return m_microcode_addr & 0xf; });
+	m_mcu->set_sx_r_cb([this]() { return m_microcode_addr & 0xf; } , "");
```

Comment:
Isn’t this a place you should be using the `NAME` macro on the lambdas so you don’t get an empty string in diagnostic messages if something goes wrong in the callback?

Reply:
Ah, didn't know that NAME could (should) be applied to lambdas too..